src/org/sosy_lab/java_smt/SolverContextFactory.java

```diff
@@ -284,6 +290,15 @@ private SolverContext generateContext0(Solvers solverToCreate)
         return PrincessSolverContext.create(
             config, shutdownNotifier, logfile, (int) randomSeed, nonLinearArithmetic);
 
+      case PRINCESS_BINARY:
```

Comment:
This is exactly the same as in the 'Princess' branch. We probably don't need a new name for the binary solver backend. Instead we could just set the option `generateSMTLIB2` to true, and then use `ProverOptions.useBinary` whenever a new ProverEnvironment is created. Maybe it would also be possible to turn this into a single option?

We may also support more binary backends in the future?